cpmpy/transformations/linearize.py

```diff
@@ -203,40 +200,16 @@ def linearize_constraint(cpm_expr):
         # Linear decomposition of alldifferent using bipartite matching
         sigma = boolvar(shape=(len(cpm_expr.args), 1 + ub - lb))
 
-        constraints = [sum(row) == 1 for row in sigma]  # Exactly one value
-        constraints += [sum(col) <= 1 for col in sigma.T]  # All diff values
+        constraints = [sum(row) == 1 for row in sigma]  # Each var has exactly one value
+        constraints += [sum(col) <= 1 for col in sigma.T]  # Each value is assigned to at most 1 variable
 
```

Comment:
In ManyWorlds, I have both: the at-most-one for non-Integer variables (string variables, e.g., for pigeonhole and sudoku) and the binary one for integer variables (I assume integer variables have a large domain. Use string variables if not (e.g., for sudoku I use the strings `"1", "2"` etc.)).

What I would do here is to check the size of the domain. If it is larger than quadratic the number of variables, go for the quadratic encoding. Otherwise, use the at-most-one?

Reply:
Yes, that is a good idea! I couldn't do that before as we did not have the `get_bounds()`

Reply:
Issue created.